integration/networking/bridge_test.go

```diff
@@ -726,6 +727,69 @@ func TestNonIPv6Network(t *testing.T) {
 	assert.Check(t, is.Equal(strings.TrimSpace(sysctlRes.Combined()), "1"))
 }
 
+// Check that starting the daemon with '--ip6tables=false' means no ip6tables
+// rules get set up for an IPv6 bridge network.
+func TestNoIP6Tables(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+	skip.If(t, testEnv.IsRootless)
+
+	ctx := setupTest(t)
+
+	testcases := []struct {
+		name        string
+		option      string
+		expIPTables bool
+	}{
+		{
+			name:        "ip6tables on",
+			option:      "--ip6tables=true",
+			expIPTables: true,
+		},
+		{
+			name:   "ip6tables off",
+			option: "--ip6tables=false",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := testutil.StartSpan(ctx, t)
+
+			d := daemon.New(t)
+			d.StartWithBusybox(ctx, t, tc.option)
+			defer d.Stop(t)
+
+			c := d.NewClientT(t)
+			defer c.Close()
+
+			const netName = "testnet"
+			const bridgeName = "testbr"
+			const subnet = "fdb3:2511:e851:34a9::/64"
+			network.CreateNoError(ctx, t, c, netName,
+				network.WithIPv6(),
+				network.WithOption("com.docker.network.bridge.name", bridgeName),
+				network.WithIPAM(subnet, "fdb3:2511:e851:34a9::1"),
+			)
+			defer network.RemoveNoError(ctx, t, c, netName)
+
+			id := container.Run(ctx, t, c, container.WithNetworkMode(netName))
+			defer c.ContainerRemove(ctx, id, containertypes.RemoveOptions{Force: true})
+
+			res, err := exec.Command("/usr/sbin/ip6tables-save").CombinedOutput()
+			assert.NilError(t, err)
+			if tc.expIPTables {
+				assert.Check(t, is.Contains(string(res), subnet))
+				assert.Check(t, is.Contains(string(res), bridgeName))
+			} else {
+				assert.Check(t, !strings.Contains(string(res), subnet),
+					fmt.Sprintf("Didn't expect to find '%s' in '%s'", subnet, string(res)))
```

Comment:
`res` is going to be a multiline string, so this error won't be easy to read. I'd rather put `... in ip6tables-save output`.

Reply:
I temporarily added a `false &&` to make the test fail - the output looks like this ...

```
=== RUN   TestNoIP6Tables/ip6tables_off
    bridge_test.go:782: assertion failed: expression is false: false && !strings.Contains(string(res), subnet): Didn't expect to find 'fdb3:2511:e851:34a9::/64' in '# Generated by ip6tables-save v1.8.9 on Tue May  7 10:34:13 2024
        *nat
        :PREROUTING ACCEPT [0:0]
        :INPUT ACCEPT [0:0]
        :OUTPUT ACCEPT [0:0]
        :POSTROUTING ACCEPT [0:0]
        :DOCKER - [0:0]
        -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
        -A OUTPUT ! -d ::1/128 -m addrtype --dst-type LOCAL -j DOCKER
        COMMIT
        # Completed on Tue May  7 10:34:13 2024
        # Generated by ip6tables-save v1.8.9 on Tue May  7 10:34:13 2024
        *filter
        :INPUT ACCEPT [0:0]
        :FORWARD DROP [0:0]
        :OUTPUT ACCEPT [2:232]
        :DOCKER - [0:0]
        :DOCKER-ISOLATION-STAGE-1 - [0:0]
        :DOCKER-ISOLATION-STAGE-2 - [0:0]
        :DOCKER-USER - [0:0]
        -A FORWARD -j DOCKER-USER
        -A FORWARD -j DOCKER-ISOLATION-STAGE-1
        -A DOCKER-ISOLATION-STAGE-1 -i docker0 ! -o docker0 -j DOCKER-ISOLATION-STAGE-2
        -A DOCKER-ISOLATION-STAGE-1 -j RETURN
        -A DOCKER-ISOLATION-STAGE-2 -o docker0 -j DROP
        -A DOCKER-ISOLATION-STAGE-2 -j RETURN
        -A DOCKER-USER -j RETURN
        COMMIT
        # Completed on Tue May  7 10:34:13 2024
        '
--- FAIL: TestNoIP6Tables (8.31s)
```

So, I think it's readable - and probably useful to see the full output.

Reply:
Okay, let's keep it as is then.